### PR TITLE
[Feature][Connector-V2][Hbase] Introduce hbase sink connector

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

#3018 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Test case:

```hocon
env {
  job.mode = "BATCH"
  execution.parallelism = 1
}

source {
  FakeSource {
    row.num = 200000
    schema {
      fields {
        name = string
        age = int
      }
    }
  }
}

transform {}

sink {
  Hbase {
    zookeeper_quorum = "hadoop001:2181,hadoop002:2181,hadoop003:2181"
    table = "seatunnel_test"
    rowkey_column = ["name"]
    family_name {
      all_columns = seatunnel
    }
  }
}

```

Zeta:

image

Spark:

image

Flink:

image

Hbase result:

image

**Comment by TyrantLucifer** [comment]
@CalvinKirs @hailin0 @EricJoy2048 @ic4y @ashulin PTAL.

File: seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkWriter.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+        // initialize hbase configuration
+        hbaseConfiguration.set("hbase.zookeeper.quorum", hbaseParameters.getZookeeperQuorum());
+        if (hbaseParameters.getHbaseExtraConfig() != null) {
+            hbaseParameters.getHbaseExtraConfig().forEach(hbaseConfiguration::set);
+        }
+        // initialize hbase connection
+        hbaseConnection = ConnectionFactory.createConnection(hbaseConfiguration);
+        // initialize hbase admin
+        hbaseAdmin = hbaseConnection.getAdmin();
```
Comment (davidzollo) [inline seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkWriter.java]: Is there any need to create `hbaseAdmin`? I didn't see any use of this field

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkWriter.java]: Thank you for your review. This parameter will be used to support save mode feature of sink connectors. But this feature is in the process of development. You can refer to https://github.com/apache/incubator-seatunnel/pull/4022 get more details. If you think at this stage we need remove useless parameters, I will delete it.

File: .github/workflows/backend.yml [hunk 2]
```diff
@@ -529,3 +529,27 @@ jobs:
           ./mvnw -T 1C -B verify -DskipUT=true -DskipIT=false -D"checkstyle.skip"=true -D"license.skipAddThirdParty"=true --no-snapshot-updates -pl $run_it_modules -am -Pci
         env:
           MAVEN_OPTS: -Xmx4096m
+
+  jdbc-connectors-it:
```
Comment (hailin0) [inline .github/workflows/backend.yml]: revert？

https://github.com/apache/incubator-seatunnel/pull/4068

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java [hunk 3]
```diff
@@ -71,8 +74,10 @@ public void startUp() throws Exception {
             .withEnv("cluster.routing.allocation.disk.threshold_enabled", "false")
             .withNetworkAliases("elasticsearch")
             .withPassword("elasticsearch")
+            .withStartupAttempts(5)
+            .withStartupTimeout(Duration.ofMinutes(5))
             .withLogConsumer(new Slf4jLogConsumer(DockerLoggerFactory.getLogger("elasticsearch:8.0.0")));
-        container.start();
+        Startables.deepStart(Stream.of(container)).join();
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java]: revert？


https://github.com/apache/incubator-seatunnel/pull/4068

File: tools/update_modules_check/update_modules_check.py [hunk 4]
```diff
@@ -150,6 +150,7 @@ def get_deleted_modules(files):
 
 def get_sub_it_modules(modules, total_num, current_num):
     modules_arr = modules.split(",")
+    modules_arr.remove("connector-jdbc-e2e")
```
Comment (hailin0) [inline tools/update_modules_check/update_modules_check.py]: revert

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake-to-hbase.conf [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    job.mode = "BATCH"
+}
+
+source {
+    FakeSource {
+        row.num = 5
+        schema {
+            fields {
+                name = string
+                age = int
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake-to-hbase.conf:29]: test all datatypes？

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+@Slf4j
+@Disabled("Hbase docker e2e case need user add mapping information of between container id and ip address in hosts file")
+public class HbaseIT extends TestSuiteBase implements TestResource {
+
+    private static final String IMAGE = "harisekhon/hbase";
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: add image version？

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                .setColumnFamily(familyDescriptor)
+                .build();
+        admin.createTable(tableDescriptor);
+        log.info("Hbase table has been initialized");
+    }
+
+    @TestTemplate
+    public void testHbaseSink(TestContainer container) throws IOException, InterruptedException {
+        Container.ExecResult execResult = container.executeJob("/fake-to-hbase.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java:130]: add data rows & fields validate

**Review by EricJoy2048 — COMMENTED** [review]
LGTM

**Review by hailin0 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**